src/runmode-af-packet.c

```diff
@@ -290,7 +295,10 @@ static void *ParseAFPConfig(const char *iface)
     if (ConfGetChildValueWithDefault(if_root, if_default, "cluster-id", &tmpclusterid) != 1) {
         aconf->cluster_id = (uint16_t)(cluster_id_auto++);
     } else {
-        aconf->cluster_id = (uint16_t)atoi(tmpclusterid);
+        if (ByteExtractStringInt32(&aconf->cluster_id, 10, 0, (const char *)tmpclusterid) < 0) {
```

Comment:
Uint16

Reply:
If `cluster_id` is defined as an `int`, this means we have a double cast :
- first an explicit cast from `int` to `uint16` where 0x23456 becomes 0x3456 (and I am not sure what 0xFFFFFFFF becomes)
- then an implicit cast from `uint16` to `int` but the value does not change since every `uint16` value can be an `int`

If you use `ByteExtractStringUint16`, do you get a warning ?
Are there other places than this where `cluster_id` can have a value which is not an `uint16` (such as e negative value) ?
Otherwise, I would change the declaration of  `cluster_id` to be `uint16`

Reply:
> If `cluster_id` is defined as an `int`, this means we have a double cast :
> 
> * first an explicit cast from `int` to `uint16` where 0x23456 becomes 0x3456 (and I am not sure what 0xFFFFFFFF becomes)
> * then an implicit cast from `uint16` to `int` but the value does not change since every `uint16` value can be an `int`
> 
Thanks for the explanation.

> If you use `ByteExtractStringUint16`, do you get a warning ?
Yes. Since `ByteExtractStringUint16` expects its first argument to be `uint16_t` but instead I'm giving `int`.

```
runmode-af-packet.c: In function ‘ParseAFPConfig’:
runmode-af-packet.c:298:37: warning: passing argument 1 of ‘ByteExtractStringUint16’ from incompatible pointer type [-Wincompatible-pointer-types]
  298 |         if (ByteExtractStringUint16(&aconf->cluster_id, 10, 0, (const char *)tmpclusterid) < 0) {
      |                                     ^~~~~~~~~~~~~~~~~~
      |                                     |
      |                                     int *
In file included from runmode-af-packet.c:58:
util-byte.h:176:39: note: expected ‘uint16_t *’ {aka ‘short unsigned int *’} but argument is of type ‘int *’
  176 | int ByteExtractStringUint16(uint16_t *res, int base, uint16_t len, const char *str);
      |                             ~~~~~~~~~~^~~
```
> Are there other places than this where `cluster_id` can have a value which is not an `uint16` (such as e negative value) ?
There aren't any.

> Otherwise, I would change the declaration of `cluster_id` to be `uint16`
Done. Thanks a lot!

Reply:
Yes, this makes sense to do including for ```AFPThreadVars::cluster_id```. Please make sure to put these af-packet changes in their own commit.